src/main/scala/com/qubole/spark/hiveacid/reader/TableReader.scala

```diff
@@ -111,19 +114,38 @@ private[hiveacid] class TableReader(sparkSession: SparkSession,
     // Create Snapshot !!!
     val curSnapshot = HiveAcidTxn.createSnapshot(curTxn, hiveAcidMetadata)
 
-    val reader = new HiveAcidReader(
+    new HiveAcidReader(
       sparkSession,
       readerOptions,
       hiveAcidReaderOptions,
-      curSnapshot.validWriteIdList)
+      curSnapshot.validWriteIdList,
+      partitions)
+  }
+
+  def getReader(requiredColumns: Array[String],
```

Comment:
I think we are returning list of Partitions here. We should not be calling it as getReader.

Reply:
makes sense, shall change it to getPartitionV2